code/game/objects/items/weapons/stunbaton.dm

```diff
@@ -141,13 +163,13 @@
 /obj/item/melee/baton/proc/baton_stun(mob/living/L, mob/user)
 	if(!ismob(L)) //because this was being called on turfs for some reason
 		return
-	
+
 	if(ishuman(L))
 		var/mob/living/carbon/human/H = L
 		if(H.check_shields(0, "[user]'s [name]", src, MELEE_ATTACK)) //No message; check_shields() handles that
 			playsound(L, 'sound/weapons/Genhit.ogg', 50, 1)
 			return
-	
+
```

Comment:
I wonder if you could put deductcharge here and have it auto shutdown when used without charge instead, by using deductcharge's return value.

Reply:
Originally when looking at solutions I had it check to see if you were a borg before applying the stun and if you didn't have the charge, shut down when you attempted to hit someone. I changed it from that because otherwise to a player it would look as though they were still able to stun someone but when they actually did so, nothing would happen.

Strangely, moving the deductcharge() to just before L.Stun() now doesn't seem to make any difference whether you're using the baton as a crew member or borg. Even if I lower my charge as a borg while the baton is still on, it will still stun and then turn off once it figures out no more power is available.